agent/reconciler/host_reconciler.go

```diff
@@ -116,12 +120,18 @@ func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastru
 
 		if r.SkipK8sInstallation {
 			logger.Info("Skipping installation of k8s components")
-		} else if r.UseInstallerController {
+		} else if r.UseInstallerController && !conditions.IsTrue(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded) {
```

Comment:
this combined condition `r.UseInstallerController && !conditions.IsTrue(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded) ` doesn't look correct to me.  

When `UseInstallerController` is set and `infrastructurev1beta1.K8sComponentsInstallationSucceeded` is marked true then on reconciliation, isn't it going inside else loop and trying to install the k8s components again?

Reply:
@mayur-tolexo So should I remove or change the condition?

Reply:
ya, fix it by using nested conditions.

Reply:
Added this. Thanks